src/util/psbt/serialize.rs

```diff
@@ -144,3 +155,172 @@ impl Deserialize for SigHashType {
         }
     }
 }
+
+// Taproot related ser/deser
+impl Serialize for schnorr::PublicKey {
+    fn serialize(&self) -> Vec<u8> {
+        self.serialize().to_vec()
+    }
+}
+
+impl Deserialize for schnorr::PublicKey {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        schnorr::PublicKey::from_slice(bytes)
+            .map_err(|_| encode::Error::ParseFailed("Invalid xonly public key"))
+    }
+}
+
+impl Serialize for (schnorrsig::Signature, SchnorrSigHashType)  {
+    fn serialize(&self) -> Vec<u8> {
+        let mut ser_sig = self.0.as_ref().to_vec();
+        if let SchnorrSigHashType::Default = self.1 {
```

Comment:
I believe exhaustive `match` would be more robust in case of refactorings.

Reply:
Do you that adding 7 lines of a match statement has value here? What sort of refactorings might that aid?

Reply:
If a new enum variant is added it'll force re-evaluating this code. I usually write something like this to limit the amount of code:

```rust
match foo {
    Foo::A => do_a(),
    Foo::B | Foo::C | Foo::D => (),
}
```